src/dxgi.rs

```diff
@@ -114,6 +118,57 @@ pub struct SwapchainDesc {
 }
 
 impl Factory2 {
+    pub fn create_swapchain_for_composition_hwnd(
```

Comment:
this seems reasonable!
Is the "_hwnd" suffix intentional here?

Reply:
It was intentional, because it also calls `IDCompositionTarget::CreateTargetForHwnd` with given hwnd. However the way this d3d12-rs crate is structured, I probably shouldn't be calling multiple things in one function?

Reply:
Yeah, I think I will refactor this so that it's just `create_swapchain_for_composition`, and it doesn't call other functions. Then I will do `comp.rs` where I move this other stuff such as creating wrappers for these:

```rust

        // Create IDCompositionTarget for the window
        let mut comp_target = CompTarget::null();
        unsafe { comp_device.CreateTargetForHwnd(hwnd as _, 1, comp_target.mut_void() as _) };

        // Create IDCompositionVisual
        let mut comp_visual = CompVisual::null();
        unsafe { comp_device.CreateVisual(comp_visual.mut_void() as _) };

        // Set swap_chain and the root visual and commit
        unsafe {
            comp_visual.SetContent(swap_chain.as_mut_ptr() as _);
            comp_target.SetRoot(comp_visual.as_mut_ptr() as _);
            comp_device.Commit();
        }
```

I move each of these in own function in `comp.rs` and I probably need to do `libloading` feature for the `DCompositionCreateDevice` function so that it's not linked.